### ISSUE TITLE
[Bug Report] hook_attn_scores masks with finfo.min instead of -inf, so isinf-based mask detection silently finds nothing

### ISSUE THREAD

**Issue by jlarson4** [TransformerLensOrg/TransformerLens#1692]
**Describe the bug**

Cached `hook_attn_scores` uses `torch.finfo(float32).min` (`-3.403e38`) for causally-masked positions, where `HookedTransformer` uses `-inf`. Unmasked values agree exactly (`0.0` max abs difference on gpt2), so the tensors are numerically equivalent for softmax purposes, but any analysis that locates the causal mask with `torch.isinf(scores)` or `scores == -torch.inf` silently matches nothing on the bridge, and the code carries on with an empty mask.

Measured on gpt2, `blocks.0.attn.hook_attn_scores`:

| | min value | `torch.isinf(...).any()` |
|---|---|---|
| HookedTransformer | `-inf` | True |
| TransformerBridge (compat mode) | `-3.403e+38` | False |

**Code example**

```python
import torch
from transformer_lens import HookedTransformer
from transformer_lens.model_bridge import TransformerBridge

ht = HookedTransformer.from_pretrained("gpt2", device="cpu", dtype=torch.float32)
br = TransformerBridge.boot_transformers("gpt2", device="cpu", dtype=torch.float32)
br.enable_compatibility_mode()

_, ch = ht.run_with_cache("The capital of France is", names_filter=["blocks.0.attn.hook_attn_scores"])
_, cb = br.run_with_cache("The capital of France is", names_filter=["blocks.0.attn.hook_attn_scores"])
s_ht, s_br = ch["blocks.0.attn.hook_attn_scores"], cb["blocks.0.attn.hook_attn_scores"]

print(torch.isinf(s_ht).any(), torch.isinf(s_br).any())   # True False
```

**System Info**

Installed from source on `dev`; macOS arm64, CPU, fp32, Python 3.12, transformers 5.13.0.

**Expected behaviour & fix pointer**

Compatibility mode should present the HookedTransformer sentinel (`-inf`) in cached `hook_attn_scores`, so that mask-detection code ported from `HookedTransformer` keeps working.

Note the sentinel is HuggingFace's, not the bridge's invention so the fix belongs at the point where the scores are exposed to the hook, not in the attention maths. If changing the cached value is judged too invasive (e.g. it would break bf16/fp16 paths or a fused kernel), the fallback is to document the sentinel in the migration guide's numerics section and in the `hook_attn_scores` docstring, since the current behaviour is a silent no-match rather than a visible difference.

Acceptance:
- [ ] Cached `hook_attn_scores` under compatibility mode uses `-inf` for masked positions, or the deviation is documented in the migration guide and the hook's docstring
- [ ] Unmasked score values remain exactly equal to HookedTransformer (currently `0.0` difference)
- [ ] Test under `tests/unit/model_bridge/` asserting the chosen sentinel
- [ ] `make unit-test` passes
- [ ] `uv run mypy .` passes

### Checklist
- [x] I have checked that there is no similar issue in the repo (**required**)

**Comment by nightcityblade** [comment]
Hi, I'd like to work on this. I'll submit a PR shortly.

**Comment by jlarson4** [comment]
@nightcityblade Assigning you now!
